pandas/core/internals.py

```diff
@@ -395,12 +395,16 @@ def fillna(self, value, limit=None, inplace=False, downcast=None,
             if not mask.any():
                 return self if inplace else self.copy()
 
-            # we cannot coerce the underlying object, so
-            # make an ObjectBlock
-            return self.to_object_block(mgr=mgr).fillna(original_value,
-                                                        limit=limit,
-                                                        inplace=inplace,
-                                                        downcast=False)
+            if errors == 'coerce' or errors == 'raise':
+                # we cannot coerce the underlying object, so
```

Comment:
Yes, we should have already caught and raised in this case, no need to have this need. Seems I made a bit of a mess while figuring out a working code path.